topics/wallet-interaction-standard.md

```diff
@@ -0,0 +1,257 @@
+// Request
+{
+    "id": 1,
+    "jsonrpc": "2.0",
+    "method": "permission",
+    "params": {
+        "version": "1",
+        "appMetadata": {
+            "name": "My DApp"
+        },
+        "networks": [{
+            "chainId": "icp:737ba355e855bd4b61279056603e0550",
+        }],
+        "scopes": ["canister_call"],
+        "challenge": "UjwgsORvEzp98TmB1cAIseNOoD9+GLyN/1DzJ5+jxZM="
+    }
+}
+
+// Response
+{
+    "id": 1,
+    "jsonrpc": "2.0",
+    "result": {
+        "version": "1",
+        "appMetadata": {
+            "name": "My Wallet"
+        },
+        "networks": [{
+            "chainId": "icp:737ba355e855bd4b61279056603e0550"
+        }],
+        "scopes": ["canister_call"],
+        "identities": [
+            {
+                "publicKey": "MFYwEAYHKoZIzj0CAQYFK4EEAAoDQgAEOTdHYwpFTr/oPXOfLQcteymk8AQE41VwPQ1W7Xpm0Zt1AY4+5aOnMAbAIjXEchxPuGbPWqPqwntXMPs3w4rOaA==" /* principal: 2mdal-aedsb-hlpnv-qu3zl-ae6on-72bt5-fwha5-xzs74-5dkaz-dfywi-aqe */
+            }
+        ],
+        "signature": "bldf7qn7DC5NzTyX5kp4GpZHaEncE5/6n/Y8av3xjEwIVFAwmhyW0uM+WBXRTj4QbScot04dfaBXUOcSWF0IjQ=="
+    }
+}
+```
+
+### `canister_call`
+
+Once the connection between the relying party and the wallet is established, and the relying party has been granted the `canister_call` permission scope, the relying party can request the wallet to execute canister calls.
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+- `nat`: A `string` value of an unsigned 64-bit integer.
+
+#### Request
+
+`version` (`text`): The version of the standard used. If the wallet does not support the version of the request, it must send the `"VERSION_NOT_SUPPORTED"` error in response.
+
+`network`: Network details on which the call should be executed.
+- `chainId` (`text`) - The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`canisterId` (`text`): The id of the canister on which the call should be executed.
+
+`sender` (`text`): The principal requested to execute the call.
+
+`method` (`text`): The name of the call method to be executed.
+
+`arg` (`blob`): The arguments for the call.
+
+
+#### Response
+
+`version` (`text`): The version of the standard used. It must match the `version` from the request.
+
+`network`: Network details on which the call was executed.
+- `chainId` (`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`contentMap`: The actual request content as specified [here](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-call).
+- `request_type` (`text`)
+- `sender` (`blob`)
```

Comment:
I've seen a mix of sender and canister id defined as both blob and text, we should probably be consistent here. In the example principals have are sent as their base32 string not as their base64 encoded bytes, so that would mean text instead of blob.